### PR TITLE
updating contributing.md based on suggestions from updates to adapter…

### PR DESCRIPTION
**Pull request by McKnight-42**
https://github.com/dbt-labs/core-team/issues/34

### Description

Updating terminology used in Contributing.md to match that of other repos to speak more explicitly. 
ex. `dbt` -> `dbt-core` 

removed references to Python 3.6.

rename `develop` to `main` in docs regarding trunk name

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -230,6 +223,8 @@ python -m pytest test/unit/test_graph.py::GraphTest::test__dependency_list
 
 dbt Labs provides a CI environment to test changes to specific adapters, and periodic maintenance checks of `dbt-core` through Github Actions. For example, if you submit a pull request to the `dbt-redshift` repo, GitHub will trigger automated code checks and tests against Redshift.
 
-A `dbt` maintainer will review your PR. They may suggest code revision for style or clarity, or request that you add unit or integration test(s). These are good things! We believe that, with a little bit of help, anyone can contribute high-quality code.
+A `dbt-core` maintainer will review your PR. They may suggest code revision for style or clarity, or request that you add unit or integration test(s). These are good things! We believe that, with a little bit of help, anyone can contribute high-quality code.
+- First time contributors should note code checks + unit tests require a maintainer to approve.
+- A maintainer will need to add the `ok to test` label, at which point your subsequent commits will trigger integration tests in BigQuery.
```
Comment (kwigley) [inline CONTRIBUTING.md:228]:
```suggestion
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,101 +20,101 @@ If you have an issue or code change suggestion related to a specific database [a
...
 
 ### Using issue labels
 
-The `dbt` maintainers use labels to categorize open issues. Some labels indicate the databases impacted by the issue, while others describe the domain in the `dbt` codebase germane to the discussion. While most of these labels are self-explanatory (eg. `snowflake` or `bigquery`), there are others that are worth describing.
+The `dbt-core` maintainers use labels to categorize open issues. Some labels indicate the databases impacted by the issue, while others describe the domain in the `dbt-core` codebase germane to the discussion. While most of these labels are self-explanatory (eg. `snowflake` or `bigquery`), there are others that are worth describing.
```
Comment (kwigley) [inline CONTRIBUTING.md:55]:
```suggestion
The `dbt-core` maintainers use labels to categorize open issues. Most labels describe the domain in the `dbt-core` codebase germane to the discussion.
```

**Review by kwigley — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -135,35 +135,37 @@ For testing, and later in the examples in this document, you may want to have `p
...
 make dev
 # or
 pip install -r dev-requirements.txt -r editable-requirements.txt
 ```
 
-When `dbt` is installed this way, any changes you make to the `dbt` source code will be reflected immediately in your next `dbt` run.
+When `dbt-core` is installed this way, any changes you make to the `dbt-core` source code will be reflected immediately in your next `dbt-core` run.
 
-### Running `dbt`
+To confirm you have correct version of dbt installed please run `dbt --version` and `which dbt`
```
Comment (kwigley) [inline CONTRIBUTING.md:152]:
```suggestion
```
Looks like we mention this in the "Running `dbt-core`" section, not sure if it is necessary.

**Review by kwigley — COMMENTED** [review]
looks good! left one suggestion

**Review by kwigley — APPROVED** [review]

**Review by leahwicz — APPROVED** [review]

**Pull request merged**